osu.Game/Overlays/ChatOverlayV2.cs

```diff
@@ -264,14 +278,21 @@ private void joinedChannelsChanged(object sender, NotifyCollectionChangedEventAr
             {
                 case NotifyCollectionChangedAction.Add:
                     IEnumerable<Channel> joinedChannels = filterChannels(args.NewItems);
+
                     foreach (var channel in joinedChannels)
                         channelList.AddChannel(channel);
+
                     break;
 
                 case NotifyCollectionChangedAction.Remove:
                     IEnumerable<Channel> leftChannels = filterChannels(args.OldItems);
+
                     foreach (var channel in leftChannels)
+                    {
                         channelList.RemoveChannel(channel);
+                        loadedChannels.Remove(channel);
```

Comment:
It may be safer to just call `.Expire()`, although that would require is to be pretty sure the chat overlay is open at this point.

Reply:
Also calling `Expire` requires the channel to be part of a parent, which means it won't work on unselected channels as well (since they're removed from the channel container). Probably best to follow the pre-existing logic mentioned above for simplicity.

Reply:
Updated to correctly dispose the drawable channel